src/Gui/QuantitySpinBox.cpp

```diff
@@ -792,49 +792,30 @@ QSize QuantitySpinBox::sizeForText(const QString& txt) const
 
 QSize QuantitySpinBox::sizeHint() const
 {
-    Q_D(const QuantitySpinBox);
-    ensurePolished();
-
-    const QFontMetrics fm(fontMetrics());
-    int h = lineEdit()->sizeHint().height();
-    int w = 0;
-
-    QString s;
-    QString fixedContent = QLatin1String(" ");
-
-    Base::Quantity q(d->quantity);
-    q.setValue(d->maximum);
-    s = textFromValue(q);
-    s.truncate(18);
-    s += fixedContent;
-    w = qMax(w, QtTools::horizontalAdvance(fm, s));
-
-    w += 2; // cursor blinking space
-    w += iconHeight;
-
-    QStyleOptionSpinBox opt;
-    initStyleOption(&opt);
-    QSize hint(w, h);
-    QSize size = style()->sizeFromContents(QStyle::CT_SpinBox, &opt, hint, this);
-    return size;
+    return sizeHintCalculator(lineEdit()->sizeHint().height());
 }
 
 QSize QuantitySpinBox::minimumSizeHint() const
+{
+    return sizeHintCalculator(lineEdit()->minimumSizeHint().height());
+}
+
+QSize QuantitySpinBox::sizeHintCalculator(int h) const
 {
     Q_D(const QuantitySpinBox);
     ensurePolished();
 
     const QFontMetrics fm(fontMetrics());
-    int h = lineEdit()->minimumSizeHint().height();
     int w = 0;
+    constexpr int maxStrLen = 12;
```

Comment:
Would it be better to declare the variable static?

Reply:
I do not know. I initially used static and read somewhere it'd be best to use consexpr but I don't know the validity of this

Reply:
`constexpr` is fine here -- the compiler will probably eliminate it completely anyway.